packages/broker/src/ConfigWizard.ts

```diff
@@ -82,200 +88,175 @@ const DefaultConfig: Config = {
-    }
+const pluginSelectorPrompt = {
+    type: 'checkbox',
+    name:'selectPlugins',
+    message: 'Select the plugins to enable',
+    choices: Object.keys(PLUGIN_TEMPLATES)
+}
 
-    async generateOrImportPrivateKey(): Promise<Wallet>{
-        const generateOrImport = await this.inquirerSinglePrompt({
-            type: 'list',
-            name:'generateOrImportEthereumPrivateKey',
-            message: 'Do you want to generate a new Ethereum private key or import an existing one?',
-            choices: ['Generate', 'Import'],
-            filter: (choice: string) => {
-                return choice.toLowerCase()
+const pluginPrompts: Array<inquirer.Question | inquirer.ListQuestion | inquirer.CheckboxQuestion> = []
+Object.keys(PLUGIN_TEMPLATES).map((pluginName) => {
+    const plugin = PLUGIN_TEMPLATES[pluginName]
+    pluginPrompts.push({
+        type: 'input',
+        name: `${pluginName}Port`,
+        message: `Select a port for the ${pluginName} Plugin [Enter for default: ${plugin.port}]`,
+        when: (answers: inquirer.Answers) => {
+            return answers.selectPlugins.includes(pluginName)
+        },
+        validate: (input: string, answers: inquirer.Answers): string | boolean => {
+            const portNumber = parseInt(input || answers[`${pluginName}Port`])
+            if (!Number.isInteger(portNumber)) {
+                return `Non-numeric value provided`
             }
-        })
 
-        let wallet: Wallet
-        if (generateOrImport === 'generate') {
-            wallet = Wallet.createRandom()
-            this.config.ethereumPrivateKey = wallet.privateKey
-            logger.info('This is your generated private key. Please store it in a secure location')
-            logger.alert(this.config.ethereumPrivateKey)
-        } else if (generateOrImport === 'import') {
-            const privateKey = await this.inquirerSinglePrompt({
-                type: 'input',
-                name: 'privateKey',
-                message: "'Please provide the private key to import'",
-            })
-
-            try {
-                wallet = new Wallet(privateKey)
-                this.config.ethereumPrivateKey = wallet.privateKey
-            } catch (privateKeyError) {
-                throw new Error(`Invalid privateKey provided for import: ${privateKey}`)
+            if (portNumber < MIN_PORT_VALUE || portNumber > MAX_PORT_VALUE) {
+                return `Out of range port ${portNumber} provided (valid range ${MIN_PORT_VALUE}-${MAX_PORT_VALUE})`
             }
 
-        } else {
-            throw new Error(`Invalid option ${generateOrImport} provided`)
+            return true
+        },
+        default: plugin.port
+    })
+})
+
+prompts = prompts.concat(pluginSelectorPrompt).concat(pluginPrompts)
+
+export const getConfigFromAnswers = (answers: any): Config => {
+    const config = { ... DEFAULT_CONFIG, plugins: { ... DEFAULT_CONFIG.plugins } }
+
+    const pluginNames = Object.keys(PLUGIN_TEMPLATES)
```

Comment:
Now this is not a template anymore, rename to PLUGIN_DEFAULT_PORTS or something? E.g.

```
{
  websocket: DEFAULT_WS_PORT
}
```